### PR TITLE
Add support for auto mode of Ja Rule

### PR DESCRIPTION
**Pull request by peternewman**
Not sure which branch I'm supposed to merge into, I still need to fix input validation too.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -124,9 +124,9 @@ AC_FUNC_STAT
 AC_FUNC_CLOSEDIR_VOID
 AC_FUNC_VPRINTF
 AC_CHECK_FUNCS([bzero gettimeofday memmove memset mkdir strdup strrchr \
-                if_nametoindex inet_ntoa inet_aton select socket strerror \
-                getifaddrs getloadavg getpwnam_r getpwuid_r getgrnam_r \
-                getgrgid_r secure_getenv])
+                if_nametoindex inet_ntoa inet_ntop inet_aton inet_pton select \
```
Comment (peternewman) [inline configure.ac]: These are ready for https://github.com/OpenLightingProject/ola/issues/788 .

**Comment by nomis52** [comment]
I think it would be much easier if we just avoid passing --enable-jarule to the pi build slaves.

I want to remove this flag as soon as I can, so there isn't much to gain by trying to make it complicated.

**Comment by peternewman** [comment]
We could do, but that involves a chunk of buildbot copy/paste, and when we remove the flag, we'll still need some logic to build Ja Rule automatically if possible (although I guess that could be in ifdefs in the code).

**Comment by nomis52** [comment]
I'd prefer to copy and paste in buildbot rather than here.

My goal is for the ja-rule code to be build if the libusb plugin is enabled and the appropriate version is met. That way we can get rid of the flag all together.

**Comment by peternewman** [comment]
Okay, understood.

**Pull request closed without merging**